WordPress/src/main/java/org/wordpress/android/ui/sitecreation/verticals/NewSiteCreationVerticalsViewHolder.kt

```diff
@@ -21,14 +22,17 @@ sealed class NewSiteCreationVerticalsViewHolder(internal val parent: ViewGroup,
     class VerticalsSuggestionItemViewHolder(
         parentView: ViewGroup
     ) : NewSiteCreationVerticalsViewHolder(parentView, R.layout.new_site_creation_verticals_suggestion_item) {
+        private val container = itemView.findViewById<ViewGroup>(R.id.container)
         private val suggestion = itemView.findViewById<TextView>(R.id.suggestion)
         private val divider = itemView.findViewById<View>(R.id.divider)
 
         override fun onBind(uiState: VerticalsListItemUiState) {
             uiState as VerticalsModelUiState
             suggestion.text = uiState.title
             divider.visibility = if (uiState.showDivider) View.VISIBLE else View.GONE
-            // TODO add onClick listener
+            container.setOnClickListener {
+                uiState.onItemTapped?.invoke() ?: AppLog.e(AppLog.T.MAIN, "onClickListener method reference is null.")
```

Comment:
Should we require `onItemTapped` to be not null at this point to follow "your" 😁 fail fast principle? My suggestion for using a `nullable` wasn't to ignore the error, but make it more obvious because I feel like `lateinit`s error is not as helpful as having a `nullable` and expecting it to be set at a certain point. Having said that I am totally fine with this approach as well, I think both approaches are better than `lateinit` for a callback, so it's your call!

Reply:
:D yep, you are probably right.... I felt like in this case it's not so crucial as it's pretty obvious it's a bug when nothing happens after you click on a list item... but we might not know about it until someone contacts support, so it's better to just crash.